new functions proposal

1ksn1ks opened the pull request:
proposal to implement 2 new functions + fix little bug with freeze inside createTopic() functions

1ksn1ks committed:
proposal for new functions

proposal for new functions

 async generatePrivateAndPublicKey()

 async updateTopic(topicId, memo, adminKey)

1ksn1ks committed:
Update hashinal-wc.es.js

1ksn1ks committed:
Update hashinal-wc.es.js

1ksn1ks committed:
Update index.d.ts

1ksn1ks committed:
Update types.d.ts

1ksn1ks committed:
import of new functions

import of new functions

1ksn1ks committed:
Update index.ts

1ksn1ks commented:
still learning

1ksn1ks committed:
update

new pull

kantorcodes commented on src/index.ts:
```diff
@@ -420,26 +422,55 @@ class HashinalsWalletConnectSDK {
+    publicKey: string;
+  }> {
+       this.ensureInitialized();
+       const privateKey = await PrivateKey.generateED25519Async();
```
Nit: the spacing in this function is slightly off

1ksn1ks replied:
fixed?

kantorcodes commented on src/index.ts:
```diff
@@ -420,26 +422,55 @@ class HashinalsWalletConnectSDK {
     }
 
-    const receipt = await this.executeTransaction(transaction);
+    const receipt = await this.executeTransaction(transaction, false); // Disable signing in executeTransaction
```
The signing should only be disabled if there's an admin key since it'll already have signed the transaction. Otherwise this function won't execute for all other cases

1ksn1ks replied:
understood

thanks

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update hashinal-wc.es.js

kantorcodes commented on src/index.ts:
```diff
@@ -420,19 +422,48 @@ class HashinalsWalletConnectSDK {
+    const privateKey = PrivateKey.fromString(adminKey);
+    const signedTx = await transaction.sign(privateKey);
+    
+    const receipt = await this.executeTransaction(signedTx);
```
Since you're already signing the transaction, the signer should be disabled in `executeTransaction`

kantorcodes commented on src/index.ts:
```diff
@@ -420,19 +422,48 @@ class HashinalsWalletConnectSDK {
       transaction.setAdminKey(adminWithPrivateKey.publicKey);
-      transaction = await transaction.sign(adminWithPrivateKey);
+      transaction.freezeWith(client); // Freeze after setting the admin key
+      transaction = await transaction.sign(adminWithPrivateKey); // Then sign
```
It looks like you're not disabling the signing in `executeTransaction`, so ultimately this transaction will still fail because its already frozen

1ksn1ks committed:
fixed

understood, makes sense

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
Update index.ts

1ksn1ks committed:
memo works

update topic memo and create topic with memo works

+ randomly generated public+privatekey also works

1ksn1ks committed:
umd updated

1ksn1ks committed:
Update current-version.txt

1ksn1ks committed:
getTopicInfo

1ksn1ks committed:
Update current-version.txt

1ksn1ks committed:
update memo

with custom fee

1ksn1ks committed:
Update current-version.txt
